overlord/devicestate/devicestate_recovery_keys_test.go

```diff
@@ -72,6 +73,9 @@ func mockSystemRecoveryKeys(c *C, alsoReinstall bool) {
 }
 
 func (s *deviceMgrRecoveryKeysSuite) TestEnsureRecoveryKeysBackwardCompat(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
```

Comment:
I don't see that you changed something that would require the locks. If indeed this is a separate change, it would fit better in a separate commit with an explanation.

If it needs to be part of the existing commit, then I think that the commit message should explain why this update was needed.

Reply:
Its a bit indirect but `deviceCtx, err := DeviceCtx(m.state, nil, nil)`in EnsureRecoveryKeys() requires the lock. The daemon sets this but our test code did not (because it was not strictly needed before).